### ISSUE TITLE
observatorium operator should support customized pull secret

### ISSUE THREAD

**Issue by marcolan018** [observatorium/operator#69]
currently observatorium CRD doesn't expose the pull secret. If the images are located in private repo and requires customized  pull secret to pull them, observatorium operator cannot help on that.
Workaround is to set the pull secret as the default one for the deploy namespace. But it's better for operator to support it so that users can specify the pull secret in the CR.

**Comment by squat** [comment]
Yes, this is a good point. This is also a good example of how the current Observatorium CRD is not well designed. It's monolithic and still not big enough! We should probably break this CRD into smaller ones that directly include Pod templates so that Observatorium does not need to explicitly handle all Kubernetes-related fields and types.
